src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -596,17 +596,22 @@ public static unsafe class PowerShellUnsafeAssemblyLoad
         /// <summary>
         /// Load an assembly in memory from unmanaged code.
         /// </summary>
-        /// <param name="data">
-        /// Unmanaged pointer to assembly data buffer.
-        /// </param>
-        /// <param name="size">
-        /// Size in bytes of the assembly data buffer.
-        /// </param>
+        /// <param name="data">Unmanaged pointer to assembly data buffer.</param>
+        /// <param name="size">Size in bytes of the assembly data buffer.</param>
+        /// <returns>Returns zero on success and non-zero on failure</returns>
         [UnmanagedCallersOnly]
-        public static void LoadAssemblyFromNativeMemory(IntPtr data, int size)
+        public static int LoadAssemblyFromNativeMemory(IntPtr data, int size)
         {
-            using var stream = new UnmanagedMemoryStream((byte*)data, size);
-            AssemblyLoadContext.Default.LoadFromStream(stream);
+            try
+            {
+                using var stream = new UnmanagedMemoryStream((byte*)data, size);
```

Comment:
We can remove `using` - LoadFromStream() does this work.

Reply:
Its `Dispose(bool disposing)` method still does something -- [setting the unsafe byte* _mem to null](https://source.dot.net/#System.Private.CoreLib/UnmanagedMemoryStream.cs,207). And I don't see the passed in `stream` object gets disposed in [AssemblyLoadContext.LoadFromStream](https://source.dot.net/#System.Private.CoreLib/AssemblyLoadContext.cs,383).